src/libs/actions/Policy.ts

```diff
@@ -872,25 +945,20 @@ function clearCustomUnitErrors(policyID, customUnitID, customUnitRateID) {
     });
 }
 
-/**
- * @param {String} policyID
- */
-function hideWorkspaceAlertMessage(policyID) {
-    if (!allPolicies[policyID]) {
+function hideWorkspaceAlertMessage(policyID: string) {
+    if (!allPolicies?.[policyID]) {
         return;
     }
 
     Onyx.merge(`${ONYXKEYS.COLLECTION.POLICY}${policyID}`, {alertMessage: ''});
 }
 
-/**
- * @param {String} policyID
- * @param {Object} currentCustomUnit
- * @param {Object} newCustomUnit
- * @param {Number} lastModified
- */
-function updateWorkspaceCustomUnitAndRate(policyID, currentCustomUnit, newCustomUnit, lastModified) {
-    const optimisticData = [
+function updateWorkspaceCustomUnitAndRate(policyID: string, currentCustomUnit: CustomUnit, newCustomUnit: NewCustomUnit, lastModified: number) {
+    if (!currentCustomUnit.customUnitID || !newCustomUnit?.customUnitID || !newCustomUnit.rates?.customUnitRateID || !currentCustomUnit.rates?.customUnitRateID) {
```

Comment:
Please make sure it works the same way as before

Reply:
It does, all are required for correct OnyxUpdate data creation so we should return when they aren't provided.